swift/README.md

```diff
@@ -1,372 +1,1100 @@
+
+### Unused Code
 
+Unused (dead) code, including Xcode template code and placeholder comments should be removed. Aspirational methods whose implementation simply calls the super class should also be removed.
+
+**Not Preferred:**
 ```swift
-let names: Array<String>  // No!
+override func didReceiveMemoryWarning() {
+   super.didReceiveMemoryWarning()
+  // Dispose of any resources that can be recreated.
+}
+
+override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+   // #warning Incomplete implementation, return the number of sections
+   return 1
+}
+
+override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+  // #warning Incomplete implementation, return the number of rows
+  return Database.contacts.count
+}
+
 ```
 
-White space
-===========
+**Preferred:**
+```swift
+override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+  return Database.contacts.count
+}
+```
+
+### Minimal Imports
+
+Keep imports minimal. For example, don't import `UIKit` when importing `Foundation` will suffice.
+
+## Spacing
+
+* Indent using 4 spaces rather than tabs to conserve space and help prevent line wrapping. Be sure to set this preference in Xcode and in the Project settings.
+* Method braces and other braces (`if`/`else`/`switch`/`while` etc.) always open on the same line as the statement but close on a new line.
+* Tip: You can re-indent by selecting some code (or ⌘A to select all) and then Control-I (or Editor\Structure\Re-Indent in the menu).
+
+**Preferred:**
+```swift
+if user.isHappy {
+  // Do something
+} else {
+  // Do something else
+}
+```
 
-Indent: 4 spaces
+**Not Preferred:**
+```swift
+if user.isHappy
+{
+  // Do something
+}
+else {
+  // Do something else
+}
+```
 
-In Xcode preferences under "Text Editing" select "Automatically trim trailing
-whitespace" and "Including whitespace-only lines".
+* There should be exactly one blank line between methods to aid in visual clarity and organization. Whitespace within methods should separate functionality, but having too many sections in a method often means you should refactor into several methods.
 
-Line lengths
-============
+* Colons always have no space on the left and one space on the right. Exceptions are the ternary operator `? :`, empty dictionary `[:]` and  `#selector` syntax for unnamed parameters `(_:)`.
 
-You are going to have long lines in Swift. Deal with it.
+**Preferred:**
+```swift
+class TestDatabase: Database {
+  var data: [String: CGFloat] = ["A": 1.2, "B": 3.2]
+}
+```
 
-Try to wrap comments to 100 characters, though.
+**Not Preferred:**
+```swift
+class TestDatabase : Database {
+  var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+}
+```
 
 For Dictionary and Array literals, use python-style indentation:
 
 ```swift
-let nameMap: [String: String] = [
+let nameMap = [
```

Comment:
Good catch!   I wonder if when we wrote this the Swift compiler wasn't smart enough to infer this.